cdap-master/src/main/java/io/cdap/cdap/master/environment/k8s/MetadataServiceMain.java

```diff
@@ -101,7 +107,15 @@ protected void configure() {
           bind(MetadataPublisher.class).to(MessagingMetadataPublisher.class);
         }
       }
-    );
+    ));
+    if (cConf.getBoolean(Constants.Security.Internal.ENFORCE_INTERNAL_AUTH)) {
```

Comment:
Does it make sense to move this block to the `SecurityModules`? It can return an empty module if auth is disabled. This would allow to reuse it

Reply:
I refactored the guice binding system in order to have both AuthenticationContextModules and ExternalAuthenticationModule dynamically install dependent modules. PTAL, thanks!

Reply:
I refactored the guice binding system in order to have both AuthenticationContextModules and ExternalAuthenticationModule dynamically install dependent modules. PTAL, thanks!